pkg/osd_cluster_upgrader/config.go

```diff
@@ -85,6 +87,9 @@ func (cfg *osdUpgradeConfig) IsValid() error {
 	if cfg.UpgradeWindow.TimeOut < 0 {
 		return fmt.Errorf("Config upgrade window time out is invalid")
 	}
+	if cfg.ExtDependencyAvailabilityCheck.HTTP.Timeout <= 0 || cfg.ExtDependencyAvailabilityCheck.HTTP.Timeout > 60 {
+		return fmt.Errorf("Config HTTP timeout is invalid (Requires int between 1 - 60 inclusive)")
+	}
```

Comment:
For robustness, I'd propose that we change this to only check it if there are actually HTTP availability URLs configured. That way, we can potentially leave out the whole `extDependencyAvailabilityChecks` from the configmap if desired, and it won't cause the controller to crashloop (its current behaviour in that scenario).

So something like:
```suggestion
	if len(cfg.ExtDependencyAvailabilityCheck.HTTP.URLS) > 0 && cfg.ExtDependencyAvailabilityCheck.HTTP.Timeout <= 0 || cfg.ExtDependencyAvailabilityCheck.HTTP.Timeout > 60 {
		return fmt.Errorf("Config HTTP timeout is invalid (Requires int between 1 - 60 inclusive)")
	}
```